### PR TITLE
Add feature usage observer.

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #.

### Summary

Add a system to observer the features used in the plugin.

### Screenshots 

----

### Test instructions

---- 

### Checklist before the final review

- [ ] Included E2E or unit tests for the changes in this PR.
- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [x] It is usable in Widgets and FSE.
- [x] Copy/Paste is working if the attributes are modified.
- [x] PR is following [the best practices]()

### CODE REVIEW

**Comment by Soare-Robert-Daniel** [comment]
### How to set up the dummy server.

ℹ️  You will need Node/Bun with MongoDB (preferably with Docker)

Server script: https://pastebin.com/1VVwEE9i | package.json script: https://pastebin.com/nAdydQHQ

Install with `npm ci` and run with `node server.js` or `bun server.js`

For mongo, open docker and run this command:

```bash
docker run -d --name tracking-db-mongo -p 27017:27017 -e MONGO_INITDB_ROOT_USERNAME=admin -e MONGO_INITDB_ROOT_PASSWORD=admin mongo
```

You will need to create a database with name `otter` and a collection with name `tracking`. If you use the [Mongo Compass](https://www.mongodb.com/products/tools/compass) app this can be done by pressing the plus button in the top-left area near the `Databases` label.

### Other info

JS tracking script: https://github.com/Codeinwp/otter-blocks/blob/03d57579444ea0dc4f997399a14c6c0d20fa5ac7/src/blocks/helpers/tracking.js

PHP tracking script: https://github.com/Codeinwp/otter-blocks/blob/03d57579444ea0dc4f997399a14c6c0d20fa5ac7/inc/Tracker.php

### JS Script usage.

Use `set` function to set an event and a key that allows you to change it. Use this when you want the event to have the last values. 

Use `add` function for things that do not change frequently or to register unique events. This function uses the `set` function, and the key is the hash of the object.

ℹ️ Consult the source code to see different options for those functions.

Use `start`, `stop`, `refresh` to start an interval for sending an event once in 5 minutes. This is not usually necessary since events are sent automatically when the user exits the page or saves the post in Post Editor.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Can you fix the PHPStan errors?

**Review by HardeepAsrani — APPROVED** [review]

File: plugins/otter-pro/inc/plugins/class-form-emails-storing.php [hunk 1]
```diff
@@ -1325,6 +1331,51 @@ public function schedule_automatic_confirmation() {
 		}
 	}
 
+	/**
```
Comment (HardeepAsrani) [inline plugins/otter-pro/inc/plugins/class-form-emails-storing.php:1334]: Also as discussed on Slack, we can remove it as this doesn't give us much info.

File: inc/Tracker.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		}
+	}
+
+	/**
+	 * Check if the user has consented to tracking.
+	 *
+	 * @return bool
+	 */
+	public static function has_consent() {
+		return boolval( get_option( 'otter_blocks_logger_flag', false ) );
```
Comment (cristian-ungureanu) [inline inc/Tracker.php:72]: '(bool) get_option( 'otter_blocks_logger_flag', false )' can be used instead (reduces cognitive load, up to 6x times faster in PHP 5.x).

**Pull request merged**